snow/networking/router/chain_router.go

```diff
@@ -51,8 +55,18 @@ type ChainRouter struct {
 	criticalChains   ids.Set
 	onFatal          func()
 	metrics          routerMetrics
+	// Parameters for doing health checks
+	healthConfig HealthConfig
 	// Unique-ified request ID --> Time and type of message made
 	requests map[ids.ID]request
+	// Counts the number of times we drop a message to any chain
+	// within a given time duration specified in [healthConfig]
+	dropMeter timer.TimedMeter
+	// Counts the number of times we successfully pass a message to any chain
+	// within a given time duration specified in [healthConfig]
+	successMeter timer.TimedMeter
```

Comment:
I don't think the current `TimedMeter` is robust enough for this. You could probably just use the EWMA utility here instead. (Success = 1, Failure = 0, percentSuccess = average)